src/data/random/financial.js

```diff
@@ -1,155 +1,127 @@
     };
+    var walk = _walk();
 
-    function calculateInterval(start, days) {
+    function granularityInYears() {
         var millisecondsPerYear = 3.15569e10;
-
-        var toDate = new Date(start.getTime());
-        toDate.setUTCDate(start.getUTCDate() + days);
-
-        return {
-            toDate: toDate,
-            years: (toDate.getTime() - start.getTime()) / millisecondsPerYear
-        };
+        return (granularity * 1000) / millisecondsPerYear;
     }
 
-    function dataGenerator(days, years) {
-
-        var prices = walk()
-            .period(years)
-            .steps(days * stepsPerDay)
-            .mu(mu)
-            .sigma(sigma)(startPrice);
-
-        var volumes = walk()
-            .period(years)
-            .steps(days)
-            .mu(0)
-            .sigma(sigma)(startVolume);
-
-        // Add random noise
-        volumes = volumes.map(function(vol) {
-            var boundedNoiseFactor = Math.min(0, Math.max(volumeNoiseFactor, 1));
-            var multiplier = 1 + (boundedNoiseFactor * (1 - 2 * Math.random()));
-            return Math.floor(vol * multiplier);
-        });
-
-        // Save the new start values
-        startPrice = prices[prices.length - 1];
-        startVolume = volumes[volumes.length - 1];
-
-        return calculateOHLC(days, prices, volumes).filter(function(d) {
-            return filter(d.date);
-        });
+    function calculateOHLC(start, price) {
+        var prices = walk
+            .period(granularityInYears())(price);
+        var datum = {
+            date: start,
+            open: prices[0],
+            high: Math.max.apply({}, prices),
+            low: Math.min.apply({}, prices),
+            close: prices[walk.steps()]
+        };
+        datum.volume = volume(datum);
+        return datum;
     }
 
-    var gen = function(days) {
-        var date = startDate,
-            remainingDays = days,
-            result = [],
-            interval;
-
-        do {
-            interval = calculateInterval(date, remainingDays);
-            result = result.concat(dataGenerator(remainingDays, interval.years));
-            date = interval.toDate;
-            remainingDays = days - result.length;
-        }
-        while (result.length < days);
-
-        return result;
+    var financial = function(numPoints) {
+        var stream = makeStream();
+        return stream.take(numPoints);
     };
 
-    gen.mu = function(x) {
-        if (!arguments.length) {
-            return mu;
-        }
-        mu = x;
-        return gen;
-    };
-    gen.sigma = function(x) {
-        if (!arguments.length) {
-            return sigma;
-        }
-        sigma = x;
-        return gen;
-    };
-    gen.startPrice = function(x) {
-        if (!arguments.length) {
-            return startPrice;
+    financial.stream = makeStream;
+
+    function makeStream() {
+        var latest;
+        function getDatum() {
+            if (!latest) {
```

Comment:
Another advantage of ```d3.time.interval``` would allow the offset to be stored, simplifying this a bit.